panel/widgets/input.py

```diff
@@ -478,14 +481,18 @@ def _convert_to_datetime(v):
             return v
         elif isinstance(v, date):
             return datetime(v.year, v.month, v.day)
+        elif hasattr(v, "astype"):
+            return v.astype('datetime64[ms]').astype(datetime)
```

Comment:
ChatGPT's explanation on why I need two astypes:
"""
When you use astype() on a numpy.datetime64 object to convert it to a Python datetime.datetime object, it might unexpectedly return an integer instead of a datetime object. This happens because astype(datetime) is actually interpreted as astype('O'), where 'O' stands for object, and it ends up converting the datetime64 to its underlying integer representation (the number of time units since a certain epoch).

To correctly convert a numpy.datetime64 object to a Python datetime.datetime object, you should explicitly import the datetime module and use datetime.datetime in the astype() method
"""